File: spring-backend/src/main/java/sg/edu/nus/iss/wellness/dto/ChatDtos.java
package sg.edu.nus.iss.wellness.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.util.List;

/**
 * DTOs for RAG chatbot endpoints.
 *
 * @author SA62 Team
 */
public final class ChatDtos {
    private ChatDtos() {
    }

    public record ChatRequest(
            @NotBlank(message = "Question cannot be empty")
            @Size(max = 1000, message = "Question must be less than 1000 characters")
            String question
    ) {
    }

    public record SourceSnippet(String title, String snippet) {
    }

    public record ChatResponse(
            Long id,
            String question,
            String answer,
            List<SourceSnippet> sources,
            String modelName,
            Instant createdAt
    ) {
    }

    public record AiChatRequest(Long userId, String question, List<RecentRecord> recentRecords) {
    }

    public record AiChatResponse(String answer, List<SourceSnippet> sources, String modelName) {
    }

    public record RecentRecord(
            String recordDate,
            double sleepHours,
            String exerciseType,
            int exerciseMinutes,
            int moodScore
    ) {
    }
}

File: dotnet-backend/Dtos/ChatDtos.cs
namespace Wellness.Backup.Api.Dtos;

/// <summary>
/// DTOs for RAG chatbot endpoints mirrored from Spring Boot.
/// </summary>
/// <remarks>@author Tiong Zhong Cheng</remarks>
public sealed record ChatRequest(string? Question);

public sealed record SourceSnippet(string Title, string Snippet);

public sealed record ChatResponse(
    long Id,
    string Question,
    string Answer,
    IReadOnlyList<SourceSnippet> Sources,
    string? ModelName,
    DateTime CreatedAt);

public sealed record AiChatRequest(long UserId, string Question, IReadOnlyList<RecentRecord> RecentRecords);

public sealed record AiChatResponse(string Answer, IReadOnlyList<SourceSnippet> Sources, string ModelName);
